keps/sig-scheduling/5007-device-attach-before-pod-scheduled/README.md

```diff
@@ -0,0 +1,1030 @@
+On the other hand, efforts toward energy efficiency are also required for the realization of a sustainable society.
+It is expected to achieve the conflicting requirements that providing higher performance and reducing power consumption simultaneously.
+Recently, a new server architecture called Composable Disaggregated Infrastructure is emerged.
+
+In a traditional server, hardware resources such as CPUs, memory, and GPUs reside within the server.
+Composable Disaggregated Infrastructure decomposes these hardware resources and makes them available as resource pools.
+We can combine these resource by software definition so that we can create custom-made servers.
+
+Composable system is composed of resource pool and Composable Manager software.
+In Resource pool all components are connected to PCIe or CXL switches.
+Composable Manager controls the switches so as to create composed baremetals by software definition.
+It has Composable API and Operator or Kubernetes may call the API.
+Once composed baremetals are created user can install any operating system or container infrastructure.
+
+This flexibility extends further with the use of fabric devices.
+Fabric devices can be used by multiple Kubernetes clusters, not just a single one.
+Each cluster expose the device as a ResourceSlice, allowing for efficient utilization of the device.
+
+In this scenario, the ResourceSlice representing a same fabric device might be selected in multiple Kubernetes clusters simultaneously.
+If the attachment fails in one cluster, the pod will remain in a failed state in kubelet.
+
+By having the scheduler wait for the fabric device to be attached, we can reschedule the pod if the attachment fails.
+This approach is superior because it avoids unnecessary waiting and allows for immediate rescheduling.
+
+### Goals
+
+1. **Enhance the DRA Scheduling Process**: 
+Implement a feature that allows the scheduling process to wait for the completion of fabric device attachment.
+This ensures that pods are only scheduled once the necessary fabric devices are successfully attached, improving reliability and efficiency.
+
+2. **Attribute Information for Fabric Devices**: 
+Add attribute information that clearly distinguishes fabric devices requiring attachment.
+This will help in accurately identifying and managing these devices within the Kubernetes environment.
+
+3. **Prioritize Device Allocation**:
+Implement a prioritization mechanism for device allocation, favoring devices directly connected to the node over attached fabric devices.
+This hierarchy ensures optimal performance and resource utilization.
+For example, the order of preference would be: Node-local devices > Attached fabric devices > Pre-attached fabric devices.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+The basic idea is the following:
+
+1. **Add a Ready Flag to ResourceClaim**:
```

Comment:
Do we need a diagram in this doc to show how "Ready" state transition with other possible states.